onnxruntime/core/providers/coreml/coreml_execution_provider.cc

```diff
@@ -36,6 +37,163 @@ CoreMLExecutionProvider::CoreMLExecutionProvider(const CoreMLOptions& options)
+    NodeIndex producer_index = producer_node_for_input ? producer_node_for_input->Index() : NodeIndex(-1);
+    if (partition_nodes_set.find(producer_index) != partition_nodes_set.end() &&
+        new_outputs.find(input->Name()) == new_outputs.end()) {
+      new_outputs.insert(input->Name());
+    }
+  }
+}
+}  // namespace
+
+bool CoreMLExecutionProvider::ProcessIncompatibleNodes(const onnxruntime::GraphViewer& graph_viewer,
+                                                       std::unordered_set<NodeIndex>& partition_nodes,
+                                                       IndexedSubGraph::MetaDef* meta_def,
+                                                       const bool is_input,
+                                                       const logging::Logger& logger) const {
+  std::unordered_set<std::string> new_names;
+  bool update_meta_def = false;
+  const size_t original_partition_size = partition_nodes.size();
+
+  // Get the appropriate vector (inputs or outputs)
+  auto& io_names = is_input ? meta_def->inputs : meta_def->outputs;
+
+  for (const auto& io_name : io_names) {
+    // Get node arg type
+    const auto* node_arg = graph_viewer.GetNodeArg(io_name);
+    int32_t type = ONNX_NAMESPACE::TensorProto_DataType_UNDEFINED;
+    if (!GetType(*node_arg, type, logger)) {
+      LOGS(logger, ERROR) << "NodeArg " << io_name << " has no type information.";
+      continue;
+    }
+
+    // Get relevant nodes (producers or consumers)
+    std::vector<const Node*> nodes;
+    if (is_input) {
+      nodes = graph_viewer.GetConsumerNodes(io_name);
+    } else {
+      const Node* producer = graph_viewer.GetProducerNode(io_name);
+      if (producer) nodes.push_back(producer);
+    }
+
+    // Process nodes based on type compatibility
+    if (!IsSupportedInputOrOutputType(type)) {
+      for (const Node* node : nodes) {
+        if (node && partition_nodes.find(node->Index()) != partition_nodes.end()) {
+          update_meta_def = true;
+          partition_nodes.erase(node->Index());
+
+          // Update the appropriate set after node removal
+          if (is_input) {
+            UpdateInputsSetAfterNodeRemoval(partition_nodes, node, new_names, graph_viewer);
+          } else {
+            UpdateOutputsSetAfterNodeRemoval(partition_nodes, node, new_names, graph_viewer);
+          }
+
+          LOGS(logger, VERBOSE) << "Removing node " << node->Name()
+                                << " from CoreML partition due to unsupported type: "
+                                << ONNX_NAMESPACE::TensorProto_DataType_Name(
+                                       (ONNX_NAMESPACE::TensorProto_DataType)type);
+        }
+      }
+    } else {
+      // Type is supported, check if nodes are in partition
+      for (const Node* node : nodes) {
+        if (node && partition_nodes.find(node->Index()) != partition_nodes.end()) {
+          new_names.insert(io_name);
```

Comment:
what if a node in the partition has one supported input and one unsupported input? would the result be different depending on whether the supported input was handled first?